arangod/IResearch/IResearchDocument.cpp

```diff
@@ -534,23 +524,15 @@ bool FieldIterator<IndexMetaStruct>::setValue(
       valueType = AnalyzerValueType::String;
     } break;
     case VPackValueType::Custom: {
-      TRI_ASSERT(!_slice.isNone());
-      if (_isDBServer) {
-        if (!_collection.empty()) {
-          _valueBuffer = getDocumentId(_collection, _slice);
-        } else {
-          LOG_TOPIC("fb53c", WARN, arangodb::iresearch::TOPIC)
-              << "Value for `_id` attribute could not be indexed for document "
-              << transaction::helpers::extractKeyFromDocument(_slice).toString()
-              << ". To recover please recreate corresponding ArangoSearch link "
-                 "'"
-              << _linkId << "'";
-          return false;
-        }
-      } else {
-        _valueBuffer = transaction::helpers::extractIdString(_trx->resolver(),
-                                                             value, _slice);
+      if (ADB_UNLIKELY(_collection.empty())) {
+        LOG_TOPIC("fb53c", WARN, arangodb::iresearch::TOPIC)
```

Comment:
2 things are unclear to me here:
1. why was the wording changing from "link" to "index"
1. why is the differentation between DB servers and other cases not necessary anymore?

Reply:
> why was the wording changing from "link" to "index"

Because we have Inverted index and ArangoSearch link here. ArangoSearch Index more clear

Reply:
> why is the differentation between DB servers and other cases not necessary anymore?

Because in both case we have collection name. We don't need to extract it from document

Reply:
Is it actually the collection name on DB servers or a shard name?

Reply:
Re "index" vs. "link" naming: "link" is still a valid name here, because we may actually deal with a link and not with an index. Or vice versa. We just don't know here. Is there probably a better name that we can use, that would include both "link" and "index"? I don't have such name at hand, but if we could find one, it could help.
However, I am fine with making the renaming change suggested in the PR here, but please note that we will then do it in the middle of a release (i.e. between 3.10.1 and 3.10.2). So please clarify with others if it's fine to do this here, or if we should only do the renaming in devel.